lib/src/source/git.dart

```diff
@@ -253,9 +302,22 @@ class GitSource extends CachedSource {
   /// by [id] on the effective ref of [id].
   ///
   /// This assumes that the canonical clone already exists.
-  Future<String> _getRev(PackageId id) {
-    return git.run(["rev-list", "--max-count=1", _getEffectiveRef(id)],
-        workingDir: _repoCachePath(id)).then((result) => result.first);
+  Future<String> _getRev(PackageId id) async {
+    var ref = _getEffectiveRef(id);
```

Comment:
`_getEffectiveRef` doesn't really make much sense in the new world. It returns `"HEAD"` if the ID doesn't specify a revision, but that's incorrect if version resolution needs to be done. I think what you want to do here is more like:

```dart
Future<String> _getRev(PackageId id) {
  var description = _getDescription(id);

  // If the id is already resolved, there's no need to rev-list to look for it.
  var resolved = description['resolved-ref'];
  if (description is Map && resolved != null) return resolved;

  // If [id] includes a version and not a ref, try "rev-list $version" and
  // "rev-list v$version". Otherwise, try "rev-list ${_getRef(id) ?? 'HEAD'}".
}
```